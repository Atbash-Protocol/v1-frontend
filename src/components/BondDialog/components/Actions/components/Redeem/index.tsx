import { Box } from '@mui/material';

export const Redeem = ({ bondID }: { bondID: string }) => {
    return <Box>Redeem {bondID}</Box>;
};
