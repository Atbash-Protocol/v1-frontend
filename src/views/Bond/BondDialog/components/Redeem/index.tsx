import { useEffect } from 'react';

import { Box, Button, Divider } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';

import Loader from 'components/Loader';
import MemoInlineMetric from 'components/Metrics/InlineMetric';
import { theme } from 'constants/theme';
import { useWeb3Context } from 'contexts/web3/web3.context';
import { selectBondsReady, selectBondItemMetrics, selectBondRedeemMetrics, selectBondQuoteResult } from 'store/modules/bonds/bonds.selector';
import { calcBondDetails, redeemBond } from 'store/modules/bonds/bonds.thunks';
import { BondMetrics } from 'store/modules/bonds/bonds.types';
import { selectIsRedeeming, selectIsRedeemingAndAutoStaking } from 'store/modules/transactions/transactions.selectors';
import { RootState } from 'store/store';

export const Redeem = ({ bondID, metrics, recipientAddress }: { bondID: string; metrics: BondMetrics; recipientAddress: string }) => {
    const dispatch = useDispatch();
    const { t } = useTranslation();
    const { bondDiscount, vestingTerm } = selectBondRedeemMetrics(metrics);

    const bondQuote = useSelector((state: RootState) => selectBondQuoteResult(state, t));
    const isClaiming = useSelector(selectIsRedeeming);
    const isClaimingAndStaking = useSelector(selectIsRedeemingAndAutoStaking);

    const {
        state: { networkID, signer },
    } = useWeb3Context();

    const bondMetrics = [
        { value: bondQuote.interestDue, metricKey: t('bond:PendingRewards') },
        { value: bondQuote.pendingPayout, metricKey: t('bond:PendingPayout') },
        { value: bondQuote.vesting, metricKey: t('bond:TimeFullyVested') },
        { value: bondDiscount, metricKey: t('ROI') },
        { value: t('common:day', { count: vestingTerm || 0 }), metricKey: t('bond:VestingTerm') },
    ].map(({ value, metricKey }, index) => <MemoInlineMetric {...{ value, metricKey }} key={index} />);

    useEffect(() => {
        if (networkID) {
            dispatch(calcBondDetails({ bondID, value: 0, networkID }));
        }
    }, [networkID]);

    const handleClaim = ({ isAutoStake }: { isAutoStake: boolean }) => {
        if (signer) {
            dispatch(
                redeemBond({
                    recipientAddress,
                    bondID,
                    isAutoStake,
                    signer,
                }),
            );
        }
    };

    return (
        <Box>
            <Box sx={{ justifyContent: 'center', display: 'flex', alignItems: 'center', flexDirection: 'column', mb: theme.spacing(2) }}>
                <Box mt={theme.spacing(2)} width={'50%'} p={theme.spacing(2)}>
                    <Button sx={{ width: '5rem' }} onClick={() => handleClaim({ isAutoStake: false })}>
                        {isClaiming ? t('common:PendingEllipsis') : t('bond:Claim')}
                    </Button>
                </Box>

                <Box mt={theme.spacing(2)} width={'50%'} p={theme.spacing(2)}>
                    <Button sx={{ width: '5rem' }} onClick={() => handleClaim({ isAutoStake: true })}>
                        {isClaimingAndStaking ? t('common:PendingEllipsis') : t('bond:ClaimAutostake')}
                    </Button>
                </Box>
            </Box>

            <Divider variant="fullWidth" textAlign="center" sx={{ borderColor: theme.palette.primary.light, marginBottom: theme.spacing(2) }} />
            <Box>{bondMetrics}</Box>
        </Box>
    );
};

const RedeemLoader = ({ bondID, recipientAddress }: { bondID: string; recipientAddress: string }) => {
    const bondsReady = useSelector(selectBondsReady);
    const metrics = useSelector((state: RootState) => selectBondItemMetrics(state, bondID));

    if (!bondsReady) return <Loader />;
    return <Redeem bondID={bondID} metrics={metrics} recipientAddress={recipientAddress} />;
};

export default RedeemLoader;
