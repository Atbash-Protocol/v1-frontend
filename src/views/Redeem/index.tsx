import { Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';

import { BCard } from 'components/BCard';

function Redeem() {
    const { t } = useTranslation();

    return (
        <BCard title={t('bond:Redeem')} zoom={true}>
            <Typography> {t('ComingSoon')} </Typography>
        </BCard>
    );
}

export default Redeem;
