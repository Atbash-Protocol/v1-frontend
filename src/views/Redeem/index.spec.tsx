import { render } from '@testing-library/react';

import Redeem from '.';

describe('Redeem', () => {
    it('renders', () => {
        const comp = render(<Redeem />);

        expect(comp).toMatchSnapshot();
    });
});
