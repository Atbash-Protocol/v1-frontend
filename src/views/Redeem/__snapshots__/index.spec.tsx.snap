// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redeem renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1vv71kp"
      >
        <div
          class="MuiBox-root css-19na2ny"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
          >
            bond:Redeem
          </h4>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
           
          ComingSoon
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1vv71kp"
    >
      <div
        class="MuiBox-root css-19na2ny"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
        >
          bond:Redeem
        </h4>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
         
        ComingSoon
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
